src/Microsoft.Framework.Runtime/Internal/ConcurrencyUtilities.cs

```diff
@@ -31,23 +31,32 @@ public static void ExecuteWithFileLocked(string filePath, Action<bool> action)
 
         public async static Task<T> ExecuteWithFileLocked<T>(string filePath, Func<bool, Task<T>> action)
         {
-            var createdNew = false;
-            var fileLock = new Semaphore(initialCount: 0, maximumCount: 1, name: FilePathToLockName(filePath),
-                createdNew: out createdNew);
-            try
+            for (var i = 0; i < 3; ++i)
             {
-                // If this lock is already acquired by another process, wait until we can acquire it
-                if (!createdNew)
+                var createdNew = false;
+                var fileLock = new Semaphore(initialCount: 0, maximumCount: 1, name: FilePathToLockName(filePath),
```

Comment:
Let's improve `FilePathToLockName` by adding some prefix to the value returned by this
```c#
return $"DNU_RESTORE_{filePath.Replace(Path.DirectorySeparatorChar, '_')}";
```
because if some other program is using the same naming convention as ours, it can be super difficult to debug.